lumicks/pylake/channel.py

```diff
@@ -264,9 +271,11 @@ def downsampled_over(self, range_list, reduce=np.mean, where="center"):
         if not isinstance(range_list, list):
             raise TypeError("Did not pass timestamps to range_list.")
 
-        assert len(range_list[0]) == 2, "Did not pass timestamps to range_list."
-        assert self._src.start < range_list[-1][1], "No overlap between range and selected channel."
-        assert self._src.stop > range_list[0][0], "No overlap between range and selected channel"
+        if len(range_list[0]) != 2:
+            raise RuntimeError("Did not pass timestamps to range_list.")
```

Comment:
I agree. It's an error on my part.

I am wondering if we're being overly restrictive here though.

I did not want to piggyback that change in this PR (since there are many small changes), but I think that for something to be `iterable`, and returning a `start` and `stop` value at each iteration would actually be sufficient for this function to work. Doing such a hard check against a `list` doesn't feel very pythonic to me.